docs/routeUrl.md

```diff
@@ -1,66 +1,90 @@
 @function can-stache-route-helpers.routeUrl {{ routeUrl(hashes) }}
 @parent can-stache-route-helpers
 
-Returns a url using [can-route.url route.url].
+@description Returns a url using [can-route.url route.url].
 
 @signature `routeUrl( hashes... [,merge] )`
 
-Calls [can-route.url route.url] with  `hashes` as its `data` argument and an
-optional `merge`.
+  Calls [can-route.url route.url] with  `hashes` as its `data` argument and an
+  optional `merge`. `routeUrl` can be used in conjunction with other helpers.
 
-This can be used on its own to create `<a>` `href`s like:
+  This can be used on its own to create `<a>` `href`s like:
 
-```html
-<a href="{{ routeUrl(page='todos' id=todo.id) }}">details</a>
-```
+  ```html
+  <mock-url></mock-url>
+  <cooking-example></cooking-example>
+  <script type="module">
+  import {Component} from "can";
+  import "//unpkg.com/mock-url@^5";
 
-Or in conjunction with other helpers:
+  Component.extend({
+    tag: "cooking-example",
+    view: `<a href='{{ routeUrl(page="recipe" id=5) }}'>{{recipe.name}}</a>`,
+    ViewModel: {
+      recipe: {
+        default() {
+          return {name: "apple pie"};
+        }
+      }
+    }
+  });
+  </script>
+  ```
+  @codepen
 
-```html
-{{ makeLink( "details", routeUrl(page='todos', true) ) }}
-```
+  This produces (with no pretty routing rules):
 
-@signature `{{ routeUrl([merge,] hashes...) }}`
+  ```html
+  <a href='#!&page=recipe&id=5'>{{recipe.name}}</a>
+  ```
 
-Passes the hashes to `route.url` and returns the result.
+  @param {can-stache/expressions/hash} [hashes...] A hash expression like `page='edit' recipeId=id`.
```

Comment:
we should mention what passing `undefined` does.  This is described in use, but should be mentioned here.